owners/src/repo.js

```diff
@@ -61,6 +66,83 @@ class Repository {
   }
 }
 
+/**
+ * Virtual in-memory repository holding owners files.
+ */
+class VirtualRepository extends Repository {
+  /**
+   * Constructor.
+   *
+   * @param {!GitHub} github GitHub API interface.
+   */
+  constructor(github) {
+    super();
+    this.github = github;
+    this.logger = github.logger;
+    /** @type {!Map<string, !VirtualFile>} */
+    this._knownFiles = new Map();
+  }
+
+  /**
+   * Fetch the latest versions of all OWNERS files.
+   */
+  async sync() {
+    await this.findOwnersFiles();
+  }
+
+  /**
+   * Read the contents of a file from the repo.
+   *
+   * @param {string} relativePath file to read.
+   * @return {string} file contents.
+   */
+  async readFile(relativePath) {
+    const file = this._knownFiles.get(relativePath);
```

Comment:
Note: Once #552 is reviewed and merged, a follow-up PR will update this to use the cache layers/cloud storage backing.